crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -162,9 +163,14 @@ fn handle_enclosed_comment<'a>(
     locator: &Locator,
 ) -> CommentPlacement<'a> {
     match comment.enclosing_node() {
-        AnyNodeRef::Parameters(arguments) => {
-            handle_parameters_separator_comment(comment, arguments, locator)
-                .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator))
```

Comment:
Calling `handle_bracketed_end_of_line_comment` here for lambda parameters (which are never parenthesized) triggered a debug assertion in that method. Similar to tuples, we should skip this check for unparenthesized cases.